go/proof.go

```diff
@@ -204,6 +204,14 @@ func (p *ExistenceProof) CheckAgainstSpec(spec *ProofSpec) error {
 	return nil
 }
 
+func keyForComparison(spec *ProofSpec, key []byte) []byte {
```

Comment:
You also will have to use this function on `getExistProofForKey` and `getNonexistProofForKey` correct?

https://github.com/cosmos/ics23/commit/473d9a546e5cbc5c645f700b83ceec3f50645ccd#diff-39a55415fc38a90b85c05989f293fda3b7ee126010cfe63838fd9b8441e47ed1R39

Reply:
Yes, good catch

Reply:
but only on `getNonexistProof`, because the lexical comparison is only relevant for non-existence proofs